### PR TITLE
TYP: Fix typing errors caused by new numpy

### PR DESCRIPTION
**Pull request by phofl**
- [ ] closes #xxxx (Replace xxxx with the Github issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

cc @mroeschke

### CODE REVIEW

File: pandas/_typing.py [hunk 1]
```diff
@@ -65,9 +65,19 @@
...
+        str,
+        bytes,
+        np.generic,
+    ]
+
     # numpy compatible types
-    NumpyValueArrayLike = Union[npt._ScalarLike_co, npt.ArrayLike]
-    NumpySorter = Optional[npt._ArrayLikeInt_co]
+    NumpyValueArrayLike = Union[ScalarLike_co, npt.ArrayLike]
+    # Name "npt._ArrayLikeInt_co" is not defined  [name-defined]
```
Comment (MarcoGorelli) [inline pandas/_typing.py:79]: Looks like the same (false positive?) issue as `npt._ScalarLike_co` - just out of interest, why redefine `ScalarLike_co ` and ignore the error for `_ArrayLikeInt_co`?

Comment (phofl) [reply pandas/_typing.py:79]: Because the scalar case was trivial while the array cases is defined using multiple private numpy typing objects

Comment (MarcoGorelli) [reply pandas/_typing.py:79]: ok clear - no objections then

**Review by MarcoGorelli — APPROVED** [review]
looks good to me

**Comment by phofl** [comment]
I think we have to backport this

File: pandas/_typing.py [hunk 2]
```diff
@@ -65,9 +65,19 @@
 
     from pandas.io.formats.format import EngFormatter
 
+    ScalarLike_co = Union[
+        int,
+        float,
+        complex,
```
Comment (twoertwein) [inline pandas/_typing.py:71]: I think we could shorten `int | float | complex` to just `complex` (see [flake8-pyi](https://github.com/PyCQA/flake8-pyi)'s Y041)

Comment (phofl) [reply pandas/_typing.py:71]: I think the union is easier to understand (at least for me), but no strong opinion

**Pull request merged**